### PR TITLE
Dependencies: add unwatch()

### PR DESCRIPTION
**Pull request by ceball**
@philippjfr, does this cover what you need?

Also, could you comment on the 4 TODOs shown below?

screen shot 2018-08-28 at 12 33 43 pm

### CODE REVIEW

File: param/parameterized.py [hunk 1]
```diff
@@ -1137,6 +1138,48 @@ def watch(self_,parameter_name,parameter_attribute=None,fn=None):
             subscribers[parameter_attribute].append(fn)
 
 
+    # TODO:
+    #
+    #   * error if not watched? (would then need a way to ask about status)
```
Comment (philippjfr) [inline param/parameterized.py]: Probably should error yes.

File: param/parameterized.py [hunk 2]
```diff
@@ -1137,6 +1138,48 @@ def watch(self_,parameter_name,parameter_attribute=None,fn=None):
...
+    #
+    #   * would it be better to have watch() return an ID which could
+    #     later be used to remove?
```
Comment (philippjfr) [inline param/parameterized.py]: Also wondered about this but think it may be too awkward to keep track of.

File: param/parameterized.py [hunk 3]
```diff
@@ -1137,6 +1138,48 @@ def watch(self_,parameter_name,parameter_attribute=None,fn=None):
...
+    #
+    #   * this method deliberately doesn't provide an easy way to
+    #     unwatch things declared at class level (i.e. can't turn off
+    #     watching for @depends(...,watch=True)). Does that make
+    #     sense?
```
Comment (philippjfr) [inline param/parameterized.py]: That seems very reasonable, imo part of the class definition and should not be messed with.

File: param/parameterized.py [hunk 4]
```diff
@@ -1115,6 +1115,7 @@ def _spec_to_obj(self_,spec):
             return MInfo(inst=inst,cls=cls,name=attr,mthd=getattr(src,attr))
 
 
+    # TODO: I think fn should probably be first arg (required). same for unwatch below
```
[the remark this answers is not in the document]
Comment (jbednar) [reply param/parameterized.py]: So you're suggesting `def watch(self_,parameter_name,fn,parameter_attribute=None):`?

[the remark this answers is not in the document]
Comment (philippjfr) [reply param/parameterized.py]: Right, optionally we could also consider supporting the special string syntax:

``watch('value: bounds', fn)``

[the remark this answers is not in the document]
Comment (ceball) [reply param/parameterized.py]: I was suggesting `def watch(self_,fn,parameter_name,parameter_attribute=None)`.

Is that too weird for you, given the name of the method?

[the remark this answers is not in the document]
Comment (philippjfr) [reply param/parameterized.py]: I'm okay with that too, I was going to say all implementations of ``watch`` in various libraries put the parameter being listened to first but ipywidgets actually matches your suggested signature, so it doesn't seem too weird.

[the remark this answers is not in the document]
Comment (ceball) [reply param/parameterized.py]: Yes, there's a TODO somewhere about accepting the 'spec' everywhere.

File: param/parameterized.py (hunk 1 above)
Comment (jlstevens) [reply param/parameterized.py]: I guess you can catch the exception if you want to unwatch something not knowing whether you've unwatched it already.

File: param/parameterized.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jlstevens) [reply param/parameterized.py]: The `watch(self_,fn,parameter_name,parameter_attribute=None)` signature seems sensible to me.

File: param/parameterized.py [hunk 5]
```diff
@@ -1129,19 +1129,25 @@ def watch(self_,parameter_name,parameter_attribute=None,fn=None):
...
-            subscribers[parameter_attribute].append(fn)
+            getattr(subscribers[parameter_attribute],action)(fn)
 
+    def watch(self_,fn,parameter_name,parameter_attribute=None):
+        self_._watch('append',fn,parameter_name,parameter_attribute)
 
+    def unwatch(self_,fn,parameter_name,parameter_attribute=None):
+        self_._watch('remove',fn,parameter_name,parameter_attribute)
+
+    # TODO: now unused?
```
Comment (ceball) [inline param/parameterized.py]: @philippjfr shall I delete subscribe() from this PR? Or do you use it?

Comment (philippjfr) [reply param/parameterized.py]: I haven't been using it, deleting it sounds good to me.

**Pull request merged**